app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,73 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @posts = Post.all
+    authorize @posts
+  end
+
+  def show
+  end
+
+  def new
+    @post = Post.new
+    authorize @post
+  end
+
+  def edit
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.tag_list.add(@tags[:tag_list].split)
+    @post.save
+    @post.user_id = session[:user_id]
```

Comment:
let's better use devise built in `current_user` helper method instead of this `session` hack